kamon-play-2.6.x/src/main/scala/kamon/play/instrumentation/WSInstrumentation.scala

```diff
@@ -22,56 +22,60 @@ import kamon.trace.{Span, SpanCustomizer}
 import kamon.util.CallingThreadExecutionContext
 import org.aspectj.lang.ProceedingJoinPoint
 import org.aspectj.lang.annotation.{Around, Aspect, Pointcut}
-import play.api.libs.ws.StandaloneWSRequest
+import play.api.libs.ws.{StandaloneWSRequest, StandaloneWSResponse, WSRequestExecutor}
 
 import scala.concurrent.Future
 
 @Aspect
 class WSInstrumentation {
 
-  @Pointcut("execution(* play.api.libs.ws.WSRequestExecutor$$anon$2.apply(..)) && args(request)")
-  def onExecuteWSRequest(request: StandaloneWSRequest): Unit = {}
+  @Pointcut("execution(* play.api.libs.ws.ahc.StandaloneAhcWSRequest.filterWSRequestExecutor(..))")
+  def createFilterWSRequestExecutor(): Unit = {}
 
-  @Around("onExecuteWSRequest(request)")
-  def aroundExecuteRequest(pjp: ProceedingJoinPoint, request: StandaloneWSRequest): Any = {
-    val currentContext = Kamon.currentContext()
-    val clientSpan = currentContext.get(Span.ContextKey)
+  @Around("createFilterWSRequestExecutor()")
+  def aroundCreatingThing(pjp: ProceedingJoinPoint): Any = {
+    val underlyingExecutor = pjp.proceed().asInstanceOf[WSRequestExecutor]
 
-    if (clientSpan.isEmpty()) pjp.proceed()
-    else {
-      val clientSpanBuilder = Kamon.buildSpan(Play.generateHttpClientOperationName(request))
-        .asChildOf(clientSpan)
-        .withMetricTag("span.kind", "client")
-        .withTag("component", "play.client.ws")
-        .withTag("http.method", request.method)
-        .withTag("http.url", request.uri.toString)
+    new WSRequestExecutor {
+      override def apply(request: StandaloneWSRequest): Future[StandaloneWSResponse] = {
+        val currentContext = Kamon.currentContext()
+        val parentSpan = currentContext.get(Span.ContextKey)
 
-      val clientRequestSpan = currentContext.get(SpanCustomizer.ContextKey)
-        .customize(clientSpanBuilder)
-        .start()
+        val clientSpanBuilder = Kamon.buildSpan(Play.generateHttpClientOperationName(request))
+          .asChildOf(parentSpan)
+          .withMetricTag("span.kind", "client")
+          .withMetricTag("component", "play.client.ws")
```

Comment:
might want to merge this first? https://github.com/kamon-io/kamon-play/pull/31

Reply:
done, merged and rebased this PR